apps/aecore/src/aec_dev_reward.erl

```diff
@@ -103,27 +151,64 @@ parse_beneficiary(BeneficiaryShareStr) ->
 
 parse_beneficiaries([]) ->
     {error, no_beneficiaries};
-parse_beneficiaries([_|_] = BeneficiarySharesStrs) ->
-    try lists:foldl(fun (Str, Acc) -> [parse_beneficiary(Str) | Acc] end,
-                    [], BeneficiarySharesStrs) of
-        Parsed ->
-            {ok, Parsed}
+parse_beneficiaries([_|_] = BeneficiarySharesProtocolsStrs) ->
+    try lists:filtermap(fun parse_beneficiary/1, BeneficiarySharesProtocolsStrs) of
+        BeneficiarySharesProtocols ->
+            {ok, [{ProtocolVsn, beneficiaries_at_protocol(ProtocolVsn, BeneficiarySharesProtocols)} ||
+                    ProtocolVsn <- protocols()]}
     catch
         error:Reason ->
             {error, Reason}
     end.
 
+beneficiaries_at_protocol(ProtocolVsn, Beneficiaries) ->
```

Comment:
feels overly careful to check for FromProtocol and TOProtocol to be undefined.
The places where you call this function already guarantee that these are integers.

I would add a type spec that they must be integers and possibly crash if not by a guard is_integer in the function clause

Reply:
The problem is that it is also called from the test function: set_beneficiaries which could include undefined as to or from parameters.

Reply:
Yes, but that is test code... you don't need to expose this function from outside (only for testing) and therefore you can adapt the test to obtain clearer code and only export for test

Reply:
The set_beneficiaries is only exported if TEST is set. It is used externally by the chain tests.